drivers/dma/dma_gd32.c

```diff
@@ -0,0 +1,637 @@
+			   size_t size)
+{
+	const struct dma_gd32_config *cfg = dev->config;
+	struct dma_gd32_data *data = dev->data;
+
+	if (ch >= cfg->channels) {
+		LOG_ERR("reload channel must be < %" PRIu32 " (%" PRIu32 ")", cfg->channels, ch);
+		return -EINVAL;
+	}
+
+	if (data->channels[ch].busy) {
+		return -EBUSY;
+	}
+
+	gd32_dma_channel_disable(cfg->reg, ch);
+
+	gd32_dma_transfer_number_config(cfg->reg, ch, size);
+
+	switch (data->channels[ch].direction) {
+	case MEMORY_TO_MEMORY:
+	case PERIPHERAL_TO_MEMORY:
+		gd32_dma_memory_address_config(cfg->reg, ch, dst);
+		gd32_dma_periph_address_config(cfg->reg, ch, src);
+		break;
+	case MEMORY_TO_PERIPHERAL:
+		gd32_dma_memory_address_config(cfg->reg, ch, src);
+		gd32_dma_periph_address_config(cfg->reg, ch, dst);
+		break;
+	}
+
+	gd32_dma_channel_enable(cfg->reg, ch);
+
+	return 0;
+}
+
+static int dma_gd32_start(const struct device *dev, uint32_t ch)
+{
+	const struct dma_gd32_config *cfg = dev->config;
+	struct dma_gd32_data *data = dev->data;
+
+	if (ch >= cfg->channels) {
+		LOG_ERR("start channel must be < %" PRIu32 " (%" PRIu32 ")", cfg->channels, ch);
+		return -EINVAL;
+	}
+
+	gd32_dma_interrupt_enable(cfg->reg, ch, DMA_CHXCTL_FTFIE | GD32_DMA_INTERRUPT_ERRORS);
+	gd32_dma_channel_enable(cfg->reg, ch);
+	data->channels[ch].busy = true;
+
+	return 0;
+}
+
+static int dma_gd32_stop(const struct device *dev, uint32_t ch)
+{
+	const struct dma_gd32_config *cfg = dev->config;
+	struct dma_gd32_data *data = dev->data;
+
+	if (ch >= cfg->channels) {
+		LOG_ERR("stop channel must be < %" PRIu32 " (%" PRIu32 ")", cfg->channels, ch);
+		return -EINVAL;
+	}
+
+	gd32_dma_interrupt_disable(cfg->reg, ch, DMA_CHXCTL_FTFIE | GD32_DMA_INTERRUPT_ERRORS);
+	gd32_dma_interrupt_flag_clear(cfg->reg, ch, DMA_FLAG_FTF | GD32_DMA_FLAG_ERRORS);
+	gd32_dma_channel_disable(cfg->reg, ch);
+	data->channels[ch].busy = false;
+
+	return 0;
+}
+
+static int dma_gd32_get_status(const struct device *dev, uint32_t ch, struct dma_status *stat)
+{
+	const struct dma_gd32_config *cfg = dev->config;
+	struct dma_gd32_data *data = dev->data;
+
+	if (ch >= cfg->channels) {
+		LOG_ERR("channel must be < %" PRIu32 " (%" PRIu32 ")", cfg->channels, ch);
+		return -EINVAL;
+	}
+
+	stat->pending_length = gd32_dma_transfer_number_get(cfg->reg, ch);
+	stat->dir = data->channels[ch].direction;
+	stat->busy = data->channels[ch].busy;
+
+	return 0;
+}
+
+static bool dma_gd32_api_chan_filter(const struct device *dev, int ch, void *filter_param)
+{
+	uint32_t filter = *((uint32_t *)filter_param);
+
+	if (!filter_param) {
+		LOG_ERR("filter_param must not be NULL");
+		return false;
+	}
+
+	return (filter & (0x1 << ch));
+}
+
+static int dma_gd32_init(const struct device *dev)
+{
+	const struct dma_gd32_config *cfg = dev->config;
+
+	rcu_periph_clock_enable(cfg->rcu_periph_clock);
+
+	for (uint32_t i = 0; i < cfg->channels; i++) {
+		gd32_dma_interrupt_disable(cfg->reg, i, DMA_CHXCTL_FTFIE | GD32_DMA_INTERRUPT_ERRORS);
```

Comment:
long lines...

```diff
-gd32_dma_interrupt_disable(cfg->reg, i, DMA_CHXCTL_FTFIE | GD32_DMA_INTERRUPT_ERRORS);
+gd32_dma_interrupt_disable(cfg->reg, i,
+                           DMA_CHXCTL_FTFIE |
+                           GD32_DMA_INTERRUPT_ERRORS);
```

Reply:
Completed all fixes.
Sorry for taking up your time, but please review.